src/OpenTK/DisplayDevice.cs

```diff
@@ -38,103 +38,74 @@ namespace OpenTK
     public class DisplayDevice
     {
         // TODO: Add properties that describe the 'usable' size of the Display, i.e. the maximized size without the taskbar etc.
-        // TODO: Does not detect changes to primary device.
-
-        private bool primary;
-        private Rectangle bounds;
-        private DisplayResolution current_resolution = new DisplayResolution();
-        private List<DisplayResolution> available_resolutions = new List<DisplayResolution>();
-        private IList<DisplayResolution> available_resolutions_readonly;
-
-        internal object Id; // A platform-specific id for this monitor
-
-        private static readonly object display_lock = new object();
-        private static DisplayDevice primary_display;
 
+        internal object id; // A platform-specific id for this monitor
         private static Platform.IDisplayDeviceDriver implementation;
 
         static DisplayDevice()
         {
             implementation = Platform.Factory.Default.CreateDisplayDeviceDriver();
         }
 
-        internal DisplayDevice()
+        internal DisplayDevice(object id)
         {
-            available_resolutions_readonly = available_resolutions.AsReadOnly();
+            this.id = id;
         }
 
-        internal DisplayDevice(DisplayResolution currentResolution, bool primary,
-            IEnumerable<DisplayResolution> availableResolutions, Rectangle bounds,
-            object id)
-            : this()
+        /// <summary>
+        /// Gets the current resolution of this instance.
+        /// </summary>
+        public DisplayResolution CurrentResolution
         {
-            // Todo: Consolidate current resolution with bounds? Can they fall out of sync right now?
-            this.current_resolution = currentResolution;
-            IsPrimary = primary;
-            this.available_resolutions.AddRange(availableResolutions);
-            #pragma warning disable 612,618
-            this.bounds = bounds == Rectangle.Empty ? currentResolution.Bounds : bounds;
-            #pragma warning restore 612,618
-            this.Id = id;
+            get
+            {
+                return implementation.GetResolution(id);
+            }
+            set
+            {
+                if (!implementation.TryChangeResolution(id, value))
+                {
+                    throw new Graphics.GraphicsModeException(
+                        String.Format("Device {0}: Failed to change resolution to {1}.", this, value));
+                }
+            }
         }
 
         /// <summary>
         /// Gets the bounds of this instance in pixel coordinates..
         /// </summary>
+        [Obsolete("This is just an adapter to CurrentResolution.Bounds.")]
```

Comment:
No need to explain what it goes to - just specify that CurrentResolution.Bounds should be used instead. Same for the others.